File: feb06/src/publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"
#include "feb06/message.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "publisher");
    ros::NodeHandle nh;
    ros::Publisher pub_loc = nh.advertise<feb06::message>("/loc_data", 10);
    ros::Rate loop_rate(1);
    
    int number = 0;
    std_msgs::Int64 msg;
    feb06::message loc;

    loc.time = 10;
    loc.position.linear.x = 0;
    loc.position.linear.y = 0;
    loc.position.angular.z = 0;
    
    while(ros::ok())
    {
        loc.time ++;
        pub_loc.publish(loc);
        loop_rate.sleep();
    }
    return 0;
}

File: feb06/src/publisher.py
#!/usr/bin/env python3 
#coding:utf-8

import rospy
from std_msgs.msg import Int64
from feb06.msg import message

def publisher():
    rospy.init_node('publisher')
    pub = rospy.Publisher('loc_data', message, queue_size = 10)
    loc = message()
    loc.position.linear.x = 10
    loc.position.linear.y = 10
    loc.position.angular.z = 10
    while not rospy.is_shutdown():
            pub.publish(loc)
            rospy.sleep(2)
            loc.time+=1
    
if __name__=='__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass